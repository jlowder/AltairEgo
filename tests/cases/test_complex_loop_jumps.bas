10 PRINT "TEST: COMPLEX LOOP JUMPS"
20 FOR I = 1 TO 2
30 PRINT "I LOOP, I="; I
40 FOR J = 1 TO 5
50 PRINT "  J LOOP, J="; J
60 IF J = 2 THEN GOTO 90
70 NEXT J
80 GOTO 110
90 PRINT "  JUMPED OUT OF J LOOP"
100 NEXT I
110 PRINT "OUTSIDE ALL LOOPS"
120 FOR K = 1 TO 2
130 PRINT "K LOOP, K="; K
140 NEXT K
150 PRINT "TEST PASSED"
160 END
