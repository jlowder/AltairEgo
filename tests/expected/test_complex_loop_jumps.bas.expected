Altair Ego: Emulating Altair BASIC 32K Rev. 3.2
OK
OK
TEST: COMPLEX LOOP JUMPS
I LOOP, I= 1
  J LOOP, J= 1
  J LOOP, J= 2
  JUMPED OUT OF J LOOP
I LOOP, I= 2
  J LOOP, J= 1
  J LOOP, J= 2
  JUMPED OUT OF J LOOP
OUTSIDE ALL LOOPS
K LOOP, K= 1
K LOOP, K= 2
TEST PASSED
OK
